third-party/unittest/googletest/include/gtest/gtest-printers.h

```diff
@@ -510,11 +510,15 @@ GTEST_API_ void PrintTo(wchar_t wc, ::std::ostream* os);
 
 GTEST_API_ void PrintTo(char32_t c, ::std::ostream* os);
 inline void PrintTo(char16_t c, ::std::ostream* os) {
-  PrintTo(ImplicitCast_<char32_t>(c), os);
+  // FIXME: the cast from char16_t to char32_t may be incorrect
+  // for a lone surrogate
+  PrintTo(static_cast<char32_t>(c), os);
```

Comment:
Modifying third-party headers makes upgrading to newer versions much harder, are we sure we want to do that? What's the long-term plan?

Reply:
https://github.com/google/googletest/issues/4762